2025-06-24 14:21:36.171 | ERROR    | app.tool.meetspot_recommender:execute:267 - 场所推荐过程中发生错误: 'list' object has no attribute 'lower'
Traceback (most recent call last):

  File "D:\all_code\MeetSpot\web_server.py", line 224, in <module>
    uvicorn.run(app, host="127.0.0.1", port=port)
    │       │   │                           └ 8000
    │       │   └ <fastapi.applications.FastAPI object at 0x00000241B08E4EC0>
    │       └ <function run at 0x00000241AAB209A0>
    └ <module 'uvicorn' from 'D:\\Python\\Lib\\site-packages\\uvicorn\\__init__.py'>

  File "D:\Python\Lib\site-packages\uvicorn\main.py", line 580, in run
    server.run()
    │      └ <function Server.run at 0x00000241AAB23600>
    └ <uvicorn.server.Server object at 0x00000241B15AFCB0>
  File "D:\Python\Lib\site-packages\uvicorn\server.py", line 66, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ None
           │       │   │    └ <function Server.serve at 0x00000241AAB236A0>
           │       │   └ <uvicorn.server.Server object at 0x00000241B15AFCB0>
           │       └ <function run at 0x00000241AA748FE0>
           └ <module 'asyncio' from 'D:\\Python\\Lib\\asyncio\\__init__.py'>
  File "D:\Python\Lib\asyncio\runners.py", line 194, in run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x00000241B15BC820>
           │      └ <function Runner.run at 0x00000241AA7AB880>
           └ <asyncio.runners.Runner object at 0x00000241B15CC050>
  File "D:\Python\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at D:\Python\Lib\site-packages\uvicorn\server.py:70> wait_for=<Futur...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x00000241AA7A9440>
           │    └ <ProactorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x00000241B15CC050>
  File "D:\Python\Lib\asyncio\base_events.py", line 707, in run_until_complete
    self.run_forever()
    │    └ <function BaseEventLoop.run_forever at 0x00000241AA7A93A0>
    └ <ProactorEventLoop running=True closed=False debug=False>
  File "D:\Python\Lib\asyncio\base_events.py", line 678, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x00000241AA7AB1A0>
    └ <ProactorEventLoop running=True closed=False debug=False>
  File "D:\Python\Lib\asyncio\base_events.py", line 2033, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x00000241AA296340>
    └ <Handle <_asyncio.TaskStepMethWrapper object at 0x00000241B00CD450>()>
  File "D:\Python\Lib\asyncio\events.py", line 89, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle <_asyncio.TaskStepMethWrapper object at 0x00000241B00CD450>()>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle <_asyncio.TaskStepMethWrapper object at 0x00000241B00CD450>()>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle <_asyncio.TaskStepMethWrapper object at 0x00000241B00CD450>()>
  File "D:\Python\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 409, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
                   └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x00000241B15CDE80>
  File "D:\Python\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 60, in __call__
    return await self.app(scope, receive, send)
                 │    │   │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x00000241B0...
                 │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
                 │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
                 │    └ <fastapi.applications.FastAPI object at 0x00000241B08E4EC0>
                 └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x00000241B15CDE80>
  File "D:\Python\Lib\site-packages\fastapi\applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
                           │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x00000241B0...
                           │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
                           └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
  File "D:\Python\Lib\site-packages\starlette\applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x00000241B0...
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x00000241B15CF230>
          └ <fastapi.applications.FastAPI object at 0x00000241B08E4EC0>
  File "D:\Python\Lib\site-packages\starlette\middleware\errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x00000241B0E8B420>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.cors.CORSMiddleware object at 0x00000241B15CF0E0>
          └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x00000241B15CF230>
  File "D:\Python\Lib\site-packages\starlette\middleware\cors.py", line 85, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x00000241B0E8B420>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x00000241B15CEF90>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x00000241B15CF0E0>
  File "D:\Python\Lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
          │                            │    │    │     │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x00000241B0E8B420>
          │                            │    │    │     │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │                            │    │    │     └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │                            │    │    └ <starlette.requests.Request object at 0x00000241B0E375C0>
          │                            │    └ <fastapi.routing.APIRouter object at 0x00000241B07ED1D0>
          │                            └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x00000241B15CEF90>
          └ <function wrap_app_handling_exceptions at 0x00000241ACC72A20>
  File "D:\Python\Lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
          │   │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          └ <fastapi.routing.APIRouter object at 0x00000241B07ED1D0>
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <bound method Router.app of <fastapi.routing.APIRouter object at 0x00000241B07ED1D0>>
          └ <fastapi.routing.APIRouter object at 0x00000241B07ED1D0>
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
          │     │      │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │     │      │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │     │      └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │     └ <function Route.handle at 0x00000241ACC940E0>
          └ APIRoute(path='/api/find_meetspot', name='find_meetspot', methods=['POST'])
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
          │    │   │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <function request_response.<locals>.app at 0x00000241B15B1BC0>
          └ APIRoute(path='/api/find_meetspot', name='find_meetspot', methods=['POST'])
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
          │                            │    │        │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │                            │    │        │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │                            │    │        └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │                            │    └ <starlette.requests.Request object at 0x00000241B0E826D0>
          │                            └ <function request_response.<locals>.app.<locals>.app at 0x00000241B0F349A0>
          └ <function wrap_app_handling_exceptions at 0x00000241ACC72A20>
  File "D:\Python\Lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
          │   │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34A40>
          │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          └ <function request_response.<locals>.app.<locals>.app at 0x00000241B0F349A0>
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
                     │ └ <starlette.requests.Request object at 0x00000241B0E826D0>
                     └ <function get_request_handler.<locals>.app at 0x00000241B15B1B20>
  File "D:\Python\Lib\site-packages\fastapi\routing.py", line 301, in app
    raw_response = await run_endpoint_function(
                         └ <function run_endpoint_function at 0x00000241ACC72200>
  File "D:\Python\Lib\site-packages\fastapi\routing.py", line 212, in run_endpoint_function
    return await dependant.call(**values)
                 │         │      └ {'request': CafeRequest(locations=['北京朝阳区', '海淀区'], keywords='咖啡馆', user_requirements='环境安静', min_rating=4.0, max_distance=30...
                 │         └ <function find_meetspot at 0x00000241B15B1620>
                 └ Dependant(path_params=[], query_params=[], header_params=[], cookie_params=[], body_params=[ModelField(field_info=Body(Pydant...

  File "D:\all_code\MeetSpot\web_server.py", line 169, in find_meetspot
    result = await recommender.execute(
                   │           └ <function CafeRecommender.execute at 0x00000241B056F4C0>
                   └ CafeRecommender(name='place_recommender', description='推荐适合多人会面的场所。\n该工具基于多个地点的位置信息，计算最佳会面地点，并推荐附近的各类场所。\n工具会生成包含地图和推荐信息的HTML...

> File "D:\all_code\MeetSpot\app\tool\meetspot_recommender.py", line 245, in execute
    recommended_places = self._rank_places(
                         │    └ <function CafeRecommender._rank_places at 0x00000241B056FBA0>
                         └ CafeRecommender(name='place_recommender', description='推荐适合多人会面的场所。\n该工具基于多个地点的位置信息，计算最佳会面地点，并推荐附近的各类场所。\n工具会生成包含地图和推荐信息的HTML...

  File "D:\all_code\MeetSpot\app\tool\meetspot_recommender.py", line 374, in _rank_places
    place_tags = place.get("tag", "").lower()
                 │     └ <method 'get' of 'dict' objects>
                 └ {'parent': [], 'distance': '200', 'pcode': '110000', 'importance': [], 'biz_ext': {'cost': [], 'rating': '4.3', 'meal_orderin...

AttributeError: 'list' object has no attribute 'lower'
2025-06-24 14:21:57.635 | INFO     | app.tool.meetspot_recommender:execute:235 - 使用 keywords '咖啡馆' 和 types '' 未找到结果，尝试仅使用 keywords 进行搜索。
2025-06-24 14:24:16.218 | ERROR    | app.tool.meetspot_recommender:execute:267 - 场所推荐过程中发生错误: 'list' object has no attribute 'lower'
Traceback (most recent call last):

  File "D:\all_code\MeetSpot\web_server.py", line 224, in <module>
    uvicorn.run(app, host="127.0.0.1", port=port)
    │       │   │                           └ 8000
    │       │   └ <fastapi.applications.FastAPI object at 0x00000241B08E4EC0>
    │       └ <function run at 0x00000241AAB209A0>
    └ <module 'uvicorn' from 'D:\\Python\\Lib\\site-packages\\uvicorn\\__init__.py'>

  File "D:\Python\Lib\site-packages\uvicorn\main.py", line 580, in run
    server.run()
    │      └ <function Server.run at 0x00000241AAB23600>
    └ <uvicorn.server.Server object at 0x00000241B15AFCB0>
  File "D:\Python\Lib\site-packages\uvicorn\server.py", line 66, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ None
           │       │   │    └ <function Server.serve at 0x00000241AAB236A0>
           │       │   └ <uvicorn.server.Server object at 0x00000241B15AFCB0>
           │       └ <function run at 0x00000241AA748FE0>
           └ <module 'asyncio' from 'D:\\Python\\Lib\\asyncio\\__init__.py'>
  File "D:\Python\Lib\asyncio\runners.py", line 194, in run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x00000241B15BC820>
           │      └ <function Runner.run at 0x00000241AA7AB880>
           └ <asyncio.runners.Runner object at 0x00000241B15CC050>
  File "D:\Python\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at D:\Python\Lib\site-packages\uvicorn\server.py:70> wait_for=<Futur...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x00000241AA7A9440>
           │    └ <ProactorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x00000241B15CC050>
  File "D:\Python\Lib\asyncio\base_events.py", line 707, in run_until_complete
    self.run_forever()
    │    └ <function BaseEventLoop.run_forever at 0x00000241AA7A93A0>
    └ <ProactorEventLoop running=True closed=False debug=False>
  File "D:\Python\Lib\asyncio\base_events.py", line 678, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x00000241AA7AB1A0>
    └ <ProactorEventLoop running=True closed=False debug=False>
  File "D:\Python\Lib\asyncio\base_events.py", line 2033, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x00000241AA296340>
    └ <Handle <_asyncio.TaskStepMethWrapper object at 0x00000241B0ECBFD0>()>
  File "D:\Python\Lib\asyncio\events.py", line 89, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle <_asyncio.TaskStepMethWrapper object at 0x00000241B0ECBFD0>()>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle <_asyncio.TaskStepMethWrapper object at 0x00000241B0ECBFD0>()>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle <_asyncio.TaskStepMethWrapper object at 0x00000241B0ECBFD0>()>
  File "D:\Python\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 409, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
                   └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x00000241B15CDE80>
  File "D:\Python\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 60, in __call__
    return await self.app(scope, receive, send)
                 │    │   │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x00000241B1...
                 │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
                 │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
                 │    └ <fastapi.applications.FastAPI object at 0x00000241B08E4EC0>
                 └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x00000241B15CDE80>
  File "D:\Python\Lib\site-packages\fastapi\applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
                           │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x00000241B1...
                           │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
                           └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
  File "D:\Python\Lib\site-packages\starlette\applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x00000241B1...
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x00000241B15CF230>
          └ <fastapi.applications.FastAPI object at 0x00000241B08E4EC0>
  File "D:\Python\Lib\site-packages\starlette\middleware\errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x00000241B0E8B380>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.cors.CORSMiddleware object at 0x00000241B15CF0E0>
          └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x00000241B15CF230>
  File "D:\Python\Lib\site-packages\starlette\middleware\cors.py", line 85, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x00000241B0E8B380>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x00000241B15CEF90>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x00000241B15CF0E0>
  File "D:\Python\Lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
          │                            │    │    │     │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x00000241B0E8B380>
          │                            │    │    │     │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │                            │    │    │     └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │                            │    │    └ <starlette.requests.Request object at 0x00000241B09E3050>
          │                            │    └ <fastapi.routing.APIRouter object at 0x00000241B07ED1D0>
          │                            └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x00000241B15CEF90>
          └ <function wrap_app_handling_exceptions at 0x00000241ACC72A20>
  File "D:\Python\Lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
          │   │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          └ <fastapi.routing.APIRouter object at 0x00000241B07ED1D0>
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <bound method Router.app of <fastapi.routing.APIRouter object at 0x00000241B07ED1D0>>
          └ <fastapi.routing.APIRouter object at 0x00000241B07ED1D0>
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
          │     │      │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │     │      │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │     │      └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │     └ <function Route.handle at 0x00000241ACC940E0>
          └ APIRoute(path='/api/find_meetspot', name='find_meetspot', methods=['POST'])
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
          │    │   │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <function request_response.<locals>.app at 0x00000241B15B1BC0>
          └ APIRoute(path='/api/find_meetspot', name='find_meetspot', methods=['POST'])
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
          │                            │    │        │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │                            │    │        │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │                            │    │        └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │                            │    └ <starlette.requests.Request object at 0x00000241B09E3E90>
          │                            └ <function request_response.<locals>.app.<locals>.app at 0x00000241B0F349A0>
          └ <function wrap_app_handling_exceptions at 0x00000241ACC72A20>
  File "D:\Python\Lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
          │   │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34A40>
          │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          └ <function request_response.<locals>.app.<locals>.app at 0x00000241B0F349A0>
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
                     │ └ <starlette.requests.Request object at 0x00000241B09E3E90>
                     └ <function get_request_handler.<locals>.app at 0x00000241B15B1B20>
  File "D:\Python\Lib\site-packages\fastapi\routing.py", line 301, in app
    raw_response = await run_endpoint_function(
                         └ <function run_endpoint_function at 0x00000241ACC72200>
  File "D:\Python\Lib\site-packages\fastapi\routing.py", line 212, in run_endpoint_function
    return await dependant.call(**values)
                 │         │      └ {'request': CafeRequest(locations=['北京朝阳区', '海淀区'], keywords='咖啡馆', user_requirements='环境安静', min_rating=4.0, max_distance=30...
                 │         └ <function find_meetspot at 0x00000241B15B1620>
                 └ Dependant(path_params=[], query_params=[], header_params=[], cookie_params=[], body_params=[ModelField(field_info=Body(Pydant...

  File "D:\all_code\MeetSpot\web_server.py", line 169, in find_meetspot
    result = await recommender.execute(
                   │           └ <function CafeRecommender.execute at 0x00000241B056F4C0>
                   └ CafeRecommender(name='place_recommender', description='推荐适合多人会面的场所。\n该工具基于多个地点的位置信息，计算最佳会面地点，并推荐附近的各类场所。\n工具会生成包含地图和推荐信息的HTML...

> File "D:\all_code\MeetSpot\app\tool\meetspot_recommender.py", line 245, in execute
    recommended_places = self._rank_places(
                         │    └ <function CafeRecommender._rank_places at 0x00000241B056FBA0>
                         └ CafeRecommender(name='place_recommender', description='推荐适合多人会面的场所。\n该工具基于多个地点的位置信息，计算最佳会面地点，并推荐附近的各类场所。\n工具会生成包含地图和推荐信息的HTML...

  File "D:\all_code\MeetSpot\app\tool\meetspot_recommender.py", line 374, in _rank_places
    place_tags = place.get("tag", "").lower()
                 │     └ <method 'get' of 'dict' objects>
                 └ {'parent': [], 'distance': '200', 'pcode': '110000', 'importance': [], 'biz_ext': {'cost': [], 'rating': '4.3', 'meal_orderin...

AttributeError: 'list' object has no attribute 'lower'
2025-06-24 14:31:22.066 | ERROR    | app.tool.meetspot_recommender:execute:267 - 场所推荐过程中发生错误: 'list' object has no attribute 'lower'
Traceback (most recent call last):

  File "D:\all_code\MeetSpot\web_server.py", line 224, in <module>
    uvicorn.run(app, host="127.0.0.1", port=port)
    │       │   │                           └ 8000
    │       │   └ <fastapi.applications.FastAPI object at 0x00000241B08E4EC0>
    │       └ <function run at 0x00000241AAB209A0>
    └ <module 'uvicorn' from 'D:\\Python\\Lib\\site-packages\\uvicorn\\__init__.py'>

  File "D:\Python\Lib\site-packages\uvicorn\main.py", line 580, in run
    server.run()
    │      └ <function Server.run at 0x00000241AAB23600>
    └ <uvicorn.server.Server object at 0x00000241B15AFCB0>
  File "D:\Python\Lib\site-packages\uvicorn\server.py", line 66, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ None
           │       │   │    └ <function Server.serve at 0x00000241AAB236A0>
           │       │   └ <uvicorn.server.Server object at 0x00000241B15AFCB0>
           │       └ <function run at 0x00000241AA748FE0>
           └ <module 'asyncio' from 'D:\\Python\\Lib\\asyncio\\__init__.py'>
  File "D:\Python\Lib\asyncio\runners.py", line 194, in run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x00000241B15BC820>
           │      └ <function Runner.run at 0x00000241AA7AB880>
           └ <asyncio.runners.Runner object at 0x00000241B15CC050>
  File "D:\Python\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at D:\Python\Lib\site-packages\uvicorn\server.py:70> wait_for=<Futur...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x00000241AA7A9440>
           │    └ <ProactorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x00000241B15CC050>
  File "D:\Python\Lib\asyncio\base_events.py", line 707, in run_until_complete
    self.run_forever()
    │    └ <function BaseEventLoop.run_forever at 0x00000241AA7A93A0>
    └ <ProactorEventLoop running=True closed=False debug=False>
  File "D:\Python\Lib\asyncio\base_events.py", line 678, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x00000241AA7AB1A0>
    └ <ProactorEventLoop running=True closed=False debug=False>
  File "D:\Python\Lib\asyncio\base_events.py", line 2033, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x00000241AA296340>
    └ <Handle <_asyncio.TaskStepMethWrapper object at 0x00000241B00CD450>()>
  File "D:\Python\Lib\asyncio\events.py", line 89, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle <_asyncio.TaskStepMethWrapper object at 0x00000241B00CD450>()>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle <_asyncio.TaskStepMethWrapper object at 0x00000241B00CD450>()>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle <_asyncio.TaskStepMethWrapper object at 0x00000241B00CD450>()>
  File "D:\Python\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 409, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
                   └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x00000241B15CDE80>
  File "D:\Python\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 60, in __call__
    return await self.app(scope, receive, send)
                 │    │   │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x00000241B0...
                 │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
                 │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
                 │    └ <fastapi.applications.FastAPI object at 0x00000241B08E4EC0>
                 └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x00000241B15CDE80>
  File "D:\Python\Lib\site-packages\fastapi\applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
                           │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x00000241B0...
                           │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
                           └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
  File "D:\Python\Lib\site-packages\starlette\applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x00000241B0...
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x00000241B15CF230>
          └ <fastapi.applications.FastAPI object at 0x00000241B08E4EC0>
  File "D:\Python\Lib\site-packages\starlette\middleware\errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x00000241B0F37E20>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.cors.CORSMiddleware object at 0x00000241B15CF0E0>
          └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x00000241B15CF230>
  File "D:\Python\Lib\site-packages\starlette\middleware\cors.py", line 85, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x00000241B0F37E20>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x00000241B15CEF90>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x00000241B15CF0E0>
  File "D:\Python\Lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
          │                            │    │    │     │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x00000241B0F37E20>
          │                            │    │    │     │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │                            │    │    │     └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │                            │    │    └ <starlette.requests.Request object at 0x00000241B0E46AF0>
          │                            │    └ <fastapi.routing.APIRouter object at 0x00000241B07ED1D0>
          │                            └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x00000241B15CEF90>
          └ <function wrap_app_handling_exceptions at 0x00000241ACC72A20>
  File "D:\Python\Lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
          │   │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          └ <fastapi.routing.APIRouter object at 0x00000241B07ED1D0>
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <bound method Router.app of <fastapi.routing.APIRouter object at 0x00000241B07ED1D0>>
          └ <fastapi.routing.APIRouter object at 0x00000241B07ED1D0>
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
          │     │      │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │     │      │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │     │      └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │     └ <function Route.handle at 0x00000241ACC940E0>
          └ APIRoute(path='/api/find_meetspot', name='find_meetspot', methods=['POST'])
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
          │    │   │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <function request_response.<locals>.app at 0x00000241B15B1BC0>
          └ APIRoute(path='/api/find_meetspot', name='find_meetspot', methods=['POST'])
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
          │                            │    │        │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │                            │    │        │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │                            │    │        └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │                            │    └ <starlette.requests.Request object at 0x00000241B0E46620>
          │                            └ <function request_response.<locals>.app.<locals>.app at 0x00000241B0F349A0>
          └ <function wrap_app_handling_exceptions at 0x00000241ACC72A20>
  File "D:\Python\Lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
          │   │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34900>
          │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          └ <function request_response.<locals>.app.<locals>.app at 0x00000241B0F349A0>
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
                     │ └ <starlette.requests.Request object at 0x00000241B0E46620>
                     └ <function get_request_handler.<locals>.app at 0x00000241B15B1B20>
  File "D:\Python\Lib\site-packages\fastapi\routing.py", line 301, in app
    raw_response = await run_endpoint_function(
                         └ <function run_endpoint_function at 0x00000241ACC72200>
  File "D:\Python\Lib\site-packages\fastapi\routing.py", line 212, in run_endpoint_function
    return await dependant.call(**values)
                 │         │      └ {'request': CafeRequest(locations=['北京朝阳区望京', '海淀中关村'], keywords='餐厅', user_requirements='商务会议', min_rating=4.0, max_distance...
                 │         └ <function find_meetspot at 0x00000241B15B1620>
                 └ Dependant(path_params=[], query_params=[], header_params=[], cookie_params=[], body_params=[ModelField(field_info=Body(Pydant...

  File "D:\all_code\MeetSpot\web_server.py", line 169, in find_meetspot
    result = await recommender.execute(
                   │           └ <function CafeRecommender.execute at 0x00000241B056F4C0>
                   └ CafeRecommender(name='place_recommender', description='推荐适合多人会面的场所。\n该工具基于多个地点的位置信息，计算最佳会面地点，并推荐附近的各类场所。\n工具会生成包含地图和推荐信息的HTML...

> File "D:\all_code\MeetSpot\app\tool\meetspot_recommender.py", line 245, in execute
    recommended_places = self._rank_places(
                         │    └ <function CafeRecommender._rank_places at 0x00000241B056FBA0>
                         └ CafeRecommender(name='place_recommender', description='推荐适合多人会面的场所。\n该工具基于多个地点的位置信息，计算最佳会面地点，并推荐附近的各类场所。\n工具会生成包含地图和推荐信息的HTML...

  File "D:\all_code\MeetSpot\app\tool\meetspot_recommender.py", line 374, in _rank_places
    place_tags = place.get("tag", "").lower()
                 │     └ <method 'get' of 'dict' objects>
                 └ {'parent': [], 'distance': '521', 'pcode': '110000', 'importance': [], 'biz_ext': {'cost': [], 'rating': '4.0', 'meal_orderin...

AttributeError: 'list' object has no attribute 'lower'
2025-06-24 14:31:24.433 | ERROR    | app.tool.meetspot_recommender:execute:267 - 场所推荐过程中发生错误: 'list' object has no attribute 'lower'
Traceback (most recent call last):

  File "D:\all_code\MeetSpot\web_server.py", line 224, in <module>
    uvicorn.run(app, host="127.0.0.1", port=port)
    │       │   │                           └ 8000
    │       │   └ <fastapi.applications.FastAPI object at 0x00000241B08E4EC0>
    │       └ <function run at 0x00000241AAB209A0>
    └ <module 'uvicorn' from 'D:\\Python\\Lib\\site-packages\\uvicorn\\__init__.py'>

  File "D:\Python\Lib\site-packages\uvicorn\main.py", line 580, in run
    server.run()
    │      └ <function Server.run at 0x00000241AAB23600>
    └ <uvicorn.server.Server object at 0x00000241B15AFCB0>
  File "D:\Python\Lib\site-packages\uvicorn\server.py", line 66, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ None
           │       │   │    └ <function Server.serve at 0x00000241AAB236A0>
           │       │   └ <uvicorn.server.Server object at 0x00000241B15AFCB0>
           │       └ <function run at 0x00000241AA748FE0>
           └ <module 'asyncio' from 'D:\\Python\\Lib\\asyncio\\__init__.py'>
  File "D:\Python\Lib\asyncio\runners.py", line 194, in run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x00000241B15BC820>
           │      └ <function Runner.run at 0x00000241AA7AB880>
           └ <asyncio.runners.Runner object at 0x00000241B15CC050>
  File "D:\Python\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at D:\Python\Lib\site-packages\uvicorn\server.py:70> wait_for=<Futur...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x00000241AA7A9440>
           │    └ <ProactorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x00000241B15CC050>
  File "D:\Python\Lib\asyncio\base_events.py", line 707, in run_until_complete
    self.run_forever()
    │    └ <function BaseEventLoop.run_forever at 0x00000241AA7A93A0>
    └ <ProactorEventLoop running=True closed=False debug=False>
  File "D:\Python\Lib\asyncio\base_events.py", line 678, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x00000241AA7AB1A0>
    └ <ProactorEventLoop running=True closed=False debug=False>
  File "D:\Python\Lib\asyncio\base_events.py", line 2033, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x00000241AA296340>
    └ <Handle <_asyncio.TaskStepMethWrapper object at 0x00000241B00CD450>()>
  File "D:\Python\Lib\asyncio\events.py", line 89, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle <_asyncio.TaskStepMethWrapper object at 0x00000241B00CD450>()>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle <_asyncio.TaskStepMethWrapper object at 0x00000241B00CD450>()>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle <_asyncio.TaskStepMethWrapper object at 0x00000241B00CD450>()>
  File "D:\Python\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 409, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
                   └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x00000241B15CDE80>
  File "D:\Python\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 60, in __call__
    return await self.app(scope, receive, send)
                 │    │   │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x00000241B0...
                 │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
                 │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
                 │    └ <fastapi.applications.FastAPI object at 0x00000241B08E4EC0>
                 └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x00000241B15CDE80>
  File "D:\Python\Lib\site-packages\fastapi\applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
                           │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x00000241B0...
                           │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
                           └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
  File "D:\Python\Lib\site-packages\starlette\applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x00000241B0...
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x00000241B15CF230>
          └ <fastapi.applications.FastAPI object at 0x00000241B08E4EC0>
  File "D:\Python\Lib\site-packages\starlette\middleware\errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x00000241B0F37CE0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.cors.CORSMiddleware object at 0x00000241B15CF0E0>
          └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x00000241B15CF230>
  File "D:\Python\Lib\site-packages\starlette\middleware\cors.py", line 85, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x00000241B0F37CE0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x00000241B15CEF90>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x00000241B15CF0E0>
  File "D:\Python\Lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
          │                            │    │    │     │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x00000241B0F37CE0>
          │                            │    │    │     │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │                            │    │    │     └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │                            │    │    └ <starlette.requests.Request object at 0x00000241B0E46AF0>
          │                            │    └ <fastapi.routing.APIRouter object at 0x00000241B07ED1D0>
          │                            └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x00000241B15CEF90>
          └ <function wrap_app_handling_exceptions at 0x00000241ACC72A20>
  File "D:\Python\Lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
          │   │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          └ <fastapi.routing.APIRouter object at 0x00000241B07ED1D0>
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <bound method Router.app of <fastapi.routing.APIRouter object at 0x00000241B07ED1D0>>
          └ <fastapi.routing.APIRouter object at 0x00000241B07ED1D0>
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 734, in app
    await route.handle(scope, receive, send)
          │     │      │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │     │      │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │     │      └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │     └ <function Route.handle at 0x00000241ACC940E0>
          └ APIRoute(path='/api/find_meetspot', name='find_meetspot', methods=['POST'])
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
          │    │   │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <function request_response.<locals>.app at 0x00000241B15B1BC0>
          └ APIRoute(path='/api/find_meetspot', name='find_meetspot', methods=['POST'])
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
          │                            │    │        │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34AE0>
          │                            │    │        │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │                            │    │        └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │                            │    └ <starlette.requests.Request object at 0x00000241B0E46620>
          │                            └ <function request_response.<locals>.app.<locals>.app at 0x00000241B0F349A0>
          └ <function wrap_app_handling_exceptions at 0x00000241ACC72A20>
  File "D:\Python\Lib\site-packages\starlette\_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
          │   │      │        └ <function wrap_app_handling_exceptions.<locals>.wrapped_app.<locals>.sender at 0x00000241B0F34900>
          │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000024...
          │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          └ <function request_response.<locals>.app.<locals>.app at 0x00000241B0F349A0>
  File "D:\Python\Lib\site-packages\starlette\routing.py", line 73, in app
    response = await f(request)
                     │ └ <starlette.requests.Request object at 0x00000241B0E46620>
                     └ <function get_request_handler.<locals>.app at 0x00000241B15B1B20>
  File "D:\Python\Lib\site-packages\fastapi\routing.py", line 301, in app
    raw_response = await run_endpoint_function(
                         └ <function run_endpoint_function at 0x00000241ACC72200>
  File "D:\Python\Lib\site-packages\fastapi\routing.py", line 212, in run_endpoint_function
    return await dependant.call(**values)
                 │         │      └ {'request': CafeRequest(locations=['北京朝阳区CBD', '北京东城区王府井'], keywords='咖啡馆', user_requirements='价格实惠', min_rating=3.5, max_dis...
                 │         └ <function find_meetspot at 0x00000241B15B1620>
                 └ Dependant(path_params=[], query_params=[], header_params=[], cookie_params=[], body_params=[ModelField(field_info=Body(Pydant...

  File "D:\all_code\MeetSpot\web_server.py", line 169, in find_meetspot
    result = await recommender.execute(
                   │           └ <function CafeRecommender.execute at 0x00000241B056F4C0>
                   └ CafeRecommender(name='place_recommender', description='推荐适合多人会面的场所。\n该工具基于多个地点的位置信息，计算最佳会面地点，并推荐附近的各类场所。\n工具会生成包含地图和推荐信息的HTML...

> File "D:\all_code\MeetSpot\app\tool\meetspot_recommender.py", line 245, in execute
    recommended_places = self._rank_places(
                         │    └ <function CafeRecommender._rank_places at 0x00000241B056FBA0>
                         └ CafeRecommender(name='place_recommender', description='推荐适合多人会面的场所。\n该工具基于多个地点的位置信息，计算最佳会面地点，并推荐附近的各类场所。\n工具会生成包含地图和推荐信息的HTML...

  File "D:\all_code\MeetSpot\app\tool\meetspot_recommender.py", line 374, in _rank_places
    place_tags = place.get("tag", "").lower()
                 │     └ <method 'get' of 'dict' objects>
                 └ {'parent': 'B0I1BBKRSH', 'distance': '444', 'pcode': '110000', 'importance': [], 'biz_ext': {'cost': [], 'rating': '4.4', 'me...

AttributeError: 'list' object has no attribute 'lower'
